smart-metastore/src/main/java/org/smartdata/metastore/MetaStore.java

```diff
@@ -1094,6 +1126,39 @@ public synchronized void updateActions(ActionInfo[] actionInfos)
     return runningActions;
   }
 
+  public List<ActionInfo> getActions(long rid, long start, long offset) throws MetaStoreException {
+    long mark = 0;
+    long count = 0;
+    List<CmdletInfo> cmdletInfos = cmdletDao.getByRid(rid);
+    List<ActionInfo> totalActions = new ArrayList<>();
+    for (CmdletInfo cmdletInfo : cmdletInfos) {
+      List<Long> aids = cmdletInfo.getAids();
+      if (mark + aids.size()>= start + 1) {
+        long i;
+        i = start - mark;
+        for (Long aid : aids) {
+          if (i > 0) {
+            i --;
+            continue;
+          }
+          if (count < offset) {
+            ActionInfo actionInfo = getActionById(aid);
+            totalActions.add(actionInfo);
+            count ++;
+          }
+          else {
+            break;
```

Comment:
This break only quit `for (Long aid : aids)` loop. You can return the result here.